test/powershell/Host/ConsoleHost.Tests.ps1

```diff
@@ -492,6 +492,85 @@ foo
             powershell -v | Should Match $psversiontable.GitCommitId
         }
     }
+
+    Context "Ambiguous arguments" {
+        It "Ambiguous argument '<testArg>' should return possible matches" -TestCases @(
+            @{testArg="-no";expectedMatches=@("-nologo","-noexit","-noprofile","-noninteractive")},
+            @{testArg="-format";expectedMatches=@("-inputformat","-outputformat")}
+        ) {
+            param($testArg, $expectedMatches)
+            $out = powershell $testArg -File foo
+            $outString = [String]::Join(",",$out)
+            foreach ($expectedMatch in $expectedMatches)
+            {
+                $outString | Should Match $expectedMatch
+            }
+        }
+    }
+}
+
+Describe "WindowStyle argument" -Tag Feature {
+    BeforeAll {
+        $ExitCodeBadCommandLineParameter = 0xFFFD0000
+        Add-Type -Name User32 -Namespace Test -MemberDefinition @"
+public static WINDOWPLACEMENT GetPlacement(IntPtr hwnd)
+{
+    WINDOWPLACEMENT placement = new WINDOWPLACEMENT();
+    placement.length = Marshal.SizeOf(placement);
+    GetWindowPlacement(hwnd, ref placement);
+    return placement;
+}
+
+[DllImport("user32.dll", SetLastError = true)]
+[return: MarshalAs(UnmanagedType.Bool)]
+public static extern bool GetWindowPlacement(
+    IntPtr hWnd, ref WINDOWPLACEMENT lpwndpl);
+
+[Serializable]
+[StructLayout(LayoutKind.Sequential)]
+public struct WINDOWPLACEMENT
+{
+    public int length;
+    public int flags;
+    public ShowWindowCommands showCmd;
+    public System.Drawing.Point ptMinPosition;
+    public System.Drawing.Point ptMaxPosition;
+    public System.Drawing.Rectangle rcNormalPosition;
+}
+
+public enum ShowWindowCommands : int
+{
+    Hidden = 0,
+    Normal = 1,
+    Minimized = 2,
+    Maximized = 3,
+}
+"@
+    }
+
+    It "-WindowStyle <WindowStyle> should work on Windows" -Skip:(!$IsWindows) -TestCases @(
```

Comment:
All tests is excluded on Unix so we could use our pattern "$PSDefaultParameterValues["it:skip"] = ..."